### DISCUSSION TITLE
Reconciler error: unhandled HTTP error 401

### DISCUSSION THREAD

**Discussion by KomanRudden** [fluxcd/flux2#3177] [General] [answered]
Hello,

Flux has been working perfectly on our DEV cluster for months. A new cluster was created and a colleague ran `flux bootstrap ...` and unfortunately forgot to specify the correct cluster. They used the existing DEV cluster rather than the new TEST cluster when using `--path=cluster/develop` flag. 

I am now noticing this error in flux logs;

`error /flux-system.flux-system - Reconciler error unable to fetch-connect to remote 'https://atc.group.net/bitbucket/scm/psmgsbb/flux.git': unhandled HTTP error 401`

running `kubectl get gitrepositories -A`:

`flux-system   flux-system         https://atc.group.net/bitbucket/scm/psmgsbb/flux.git        253d   False   failed to checkout and determine revision: unable to list remote for 'https://atc.group.net/bitbucket/scm/psmgsbb/flux.git': authentication required`

Do I need to rerun `flux bootstrap...` on the DEV cluster to "reset" everything? What other options do I have please?

Thank you

**Comment by hiddeco** [comment]
The credentials are controlled using a Secret in the `flux-system` namespace, which can be replaced with other credentials which do have access to the repository.

However, since the configuration seems to have been rewritten in an unexpected way. I would recommend deleting the Secret and running `flux bootstrap ...` with the correct flags again to ensure all configuration is restored into the right state.

**Reply by KomanRudden** [reply]
Thank you for the reponse. I tried as you suggested. I deleted the flux-system secret and reran bootstrap. Everything succeeded and I now see this:

`kubectl get gitrepositories -A` has no errors now

`flux-system   flux-system         ssh://git@git.group.net:7999/psmgsbb/flux.git               253d   True    stored artifact for revision 'master/252f6416c034bb67f06cc3e413e66704bc6b1069'`

however I am seeing these errors now

`error ImagePolicy/post-processing-master-branch-policy.flux-system - Reconciler error Cannot determine latest tag for policy: version list argument cannot be empty`

`error HelmRelease/post-processing.post-processing-dev - Reconciler error previous release attempt remediation failed`

`error ImageRepository/post-processing-repository.flux-system - Reconciler error auth for "myacr.azurecr.io" not found in secret flux-system/psbombb-image-acr-auth-cc8mg5tk84`

Regarding the secret above I ran:

`k get secret -n flux-system psbombb-image-acr-auth-cc8mg5tk84 -oyaml`
which gave me
`apiVersion: v1
data:
  .dockerconfigjson: ewoJImRhdGEiOiAie1xuICBcImF1dGhzXCI6IHtcbiAgICBcImFjcnBzYm9tYmJha3NkZXYuYXp1cmVjci5pb1wiOiB7XG4gICAgICBcImF1dGhcIjogXCJZVE5s<changed>0ySXpMVGt5TWpndE1qQTBabVEyT0RsbE1EUmxPbXQzUVhWS2IwOU9SWGxXU0ZCNmFITmlhbEY0VjFoSlp3PT1cIlxuICAgIH1cbiAgfVxufVxuIgp9
kind: Secret
`
which decodes to 
`"data": "{\n  \"auths\": {\n    \"myacr.azurecr.io\": {\n      \"auth\": \"YTNlMTNlOGItYWQwNi00M2IzLTkyMjgtMjA0ZmQ2ODllMD<changed>09ORXlWSFB6aHNialF4V1hJZw==\"\n    }\n  }\n}\n"`

The container registry matches -> `myacr.azurecr.io`

Any further thoughts would be most welcome. Thank you again

**Comment by KomanRudden** [comment, marked as answer]
The solution to my problem was to downgrade the kustomize controller to v0.22.3. Once this was done the correct formatting was used in the dockerconfigjson file and sync'ing started again. The `bootstrap` must have upgrade the versions.
